setup.py

```diff
@@ -97,23 +101,23 @@ def build_extensions(self):
             opts.append(cpp_flag(self.compiler))
             if has_flag(self.compiler, '-fvisibility=hidden'):
                 opts.append('-fvisibility=hidden')
-            # check that native flag is available
-            native_flag = '-march=native'
-            print('checking avalability of flag:', native_flag)
-            if not has_flag(self.compiler, native_flag):
-                print('removing unsupported compiler flag:', native_flag)
-                opts.remove(native_flag)
-                # for macos add apple-m1 flag if it's available
-                if sys.platform == 'darwin':
-                    m1_flag = '-mcpu=apple-m1'
-                    print('checking avalability of flag:', m1_flag)
-                    if has_flag(self.compiler, m1_flag):
-                        print('adding flag:', m1_flag)
-                        opts.append(m1_flag)
-                    else:
-                        print(f'flag: {m1_flag} is not available')
-            else:
-                print(f'flag: {native_flag} is available')
+            if not os.environ.get("HNSWLIB_NO_NATIVE"):
+                # check that native flag is available
+                print('checking avalability of flag:', self.native_flag)
```

Comment:
I think it `BuildExt.native_flag` is better than `self.native_flag` for readability (as `self.native_flag` is never set)